xdsl/traits.py

```diff
@@ -264,6 +264,31 @@ def lookup_symbol(
                 return SymbolTable.lookup_symbol(o, nested_name)
         return None
 
+    @staticmethod
+    def insert_symbol_if_not_exists(symbol_table_op: Operation, symbol_op: Operation):
```

Comment:
3 concerns here:
1. MLIR has also a name uniquing functionality here (not very nice IMHO, but depends on the use cases)
2. Based on this, MLIR has a TODO to rename this method to `insertOrUpdate`
3. If we don't do this, maybe just name the method `insert` for now?

Reply:
I could live with renaming this `insert_or_update` and changing the functionality accordingly. Any concerns?

Reply:
You're telling me this is the name of the function in MLIR? I'd much rather go with `insert_or_update`. I'd usually call that `upsert` but not everyone is familiar with that term. Also please add a doc comment to explain the logic.

Reply:
I renamed it to `insert_or_update` to follow MLIR, and added a docstring. LMK if there's anything else that needs changing.